core/src/main/java/org/infinispan/persistence/manager/PersistenceManagerImpl.java

```diff
@@ -278,7 +297,16 @@ public void disableStore(String storeType) {
             storesMutex.writeLock().unlock();
          }
 
-         if (loaders.isEmpty() && nonTxWriters.isEmpty() && txWriters.isEmpty()) {
+         boolean noMoreStores;
+
+         storesMutex.readLock().lock();
+         try {
+            noMoreStores = loaders.isEmpty() && nonTxWriters.isEmpty() && txWriters.isEmpty();
```

Comment:
Hrmm might as well do this in the write lock above?

Reply:
:+1: